dask/array/core.py

```diff
@@ -2249,6 +2258,120 @@ def to_zarr(arr, url, component=None, storage_options=None,
                      return_stored=return_stored)
 
 
+def from_tiledb(uri, attribute=None, chunks=None,
+                storage_options=None, **kwargs):
+    """Load array from the TileDB storage database
+
+    See https://docs.tiledb.io for more information.
+
+    Parameters
+    ----------
+    uri: TileDB array or str
+        Location of the data
+    """
+    def tiledb_to_chunks(array):
+        """
```

Comment:
This docstring will not be seen by anyone. Should this info be in the main docstring, above?

Reply:
It's also nice to avoid placing dynamically created functions in task graphs.  They're slightly more expensive to serialize and ship between workers.

To resolve this I recommend that we move this function to top level, perhaps with an underscore in front of it to denote that it is not for public use.

Additionally, we might consider moving the `to_tiledb` code to a separate `dask/array/tiledb.py` file (probably the same should be done with zarr at some point.) . This would require an extra `from .tiledb import to_tiledb` in the `Array.to_tiledb` method, which would obviously have to stay in core.py.

Reply:
> To resolve this I recommend that we move this function to top level, perhaps with an underscore in front of it to denote that it is not for public use.

done

> Additionally, we might consider moving the to_tiledb code to a separate dask/array/tiledb.py file (probably the same should be done with zarr at some point.) . This would require an extra from .tiledb import to_tiledb in the Array.to_tiledb method, which would obviously have to stay in core.py.

done -- I wasn't sure if you wanted both `from_` and `to_` moved to a file, but it seems to make sense to keep them together, so I did that for now